designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala

```diff
@@ -37,4 +38,20 @@ class DefinitionResources(
     }
   }
 
+  private val isFragmentParam: Directive1[Boolean] = parameter(Symbol("isFragment").as[Boolean])
+
+  private val modelParametersModeParam: Directive1[ModelParametersMode] = {
```

Comment:
1. This `RAW` mode is a kind of quirk for external project purpose. I would document it - see comments containing "external project" and try to have it in as fewer places as possible. We can make FE don't know about the existence of this parameter at all. To achieve that, in the API we could replace it with some boolean that on the BE side will be immediately replaced by this enum.

2. Also, I would change the naming a bit - RAW/ENRICHED sounds too generic for me. We need to use `UiConfig`/`ComponentUiConfig` in the name as it is used everywhere where we touch this "customization" feature. Also, the variant that you've called "RAW" isn't raw. By "RAW" I understand something not for final usage, sth that after some additional processing will be usable and in our usages, in most cases no AdditionalUIConfig will be applied so basically both modes will return the same value.

BTW if you need some inspiration, here is the documentation: https://nussknacker.io/documentation/docs/configuration/model/ModelConfiguration/#configuration-of-ui-attributes-of-components